Fix clang-format file to position braces and parameters correctly

ssanadhya opened the pull request:
Summary:
We would like to follow the prevalent C++ style guidelines in our C/C++ code
formatting. This change fixes the clang-format style file as follows:
- In class/funtion definitions, put opening brace on the same line as the
  function/class name
- In function declarations/definitions/calls pack the parameter/arguments in as
  few lines as possible
- In class definitions, break line before the colon if the inheritance list
  doesn't fit in one line
- In constructor initialization, break line before the colon if the initializers
  don't fit in one line
- Other minor horizontal whitespace fixes
This is a preview diff to highlight the changes. Future diffs will apply the
format to entire oai/ directory.

Differential Revision: D20499279

rdefosse commented:
Hi Shruti,

nice effort.

For me it is OK. 

The only thing I don't like is the `80` characters limit per line. For me, it's a bit out-dated.
We all have now widescreens w/ high resolution.
But it is a personal opinion.

Regards
Raphael

lionelgo commented:
Very fine for me:
IndentWidth: 2
ColumnLimit: 80
IncludeBlocks: Preserve
SortIncludes: false
AlignOperands: true
DerivePointerAlignment: false
AllowAllParametersOfDeclarationOnNextLine: false
AllowShortFunctionsOnASingleLine: Inline
AlwaysBreakAfterReturnType: None
IndentWrappedFunctionNames: false
KeepEmptyLinesAtTheStartOfBlocks: false
BreakBeforeBinaryOperators: None
BreakBeforeBraces: Attach
CompactNamespaces: false
MaxEmptyLinesToKeep: 1
UseTab: Never
SpaceBeforeAssignmentOperators: true
SpaceBeforeParens: ControlStatements
SpaceBeforeRangeBasedForLoopColon: true
SpaceInEmptyParentheses: false
SpacesInAngles: false
SpacesInCStyleCastParentheses: false
SpacesInParentheses: false
SpacesInSquareBrackets: false
ConstructorInitializerAllOnOneLineOrOnePerLine: true
ConstructorInitializerIndentWidth: 4
IndentCaseLabels: true
Cpp11BracedListStyle: true // check consistency with SpacesInParentheses ?
FixNamespaceComments: true
NamespaceIndentation: None


Fine for me:
AlignConsecutiveDeclarations: false
AlignTrailingComments: true
ReflowComments: true
BreakConstructorInitializers: BeforeColon
SortUsingDeclarations: true

Equal:
AlignEscapedNewlines: Right
AlwaysBreakTemplateDeclarations: Yes
AllowShortBlocksOnASingleLine: false
BreakBeforeTernaryOperators: False
SpaceAfterCStyleCast: true
SpaceAfterTemplateKeyword: false
SpaceBeforeInheritanceColon: true
SpacesBeforeTrailingComments: 1 // had put 2 (google clang)
SpacesInContainerLiterals: false // had put true
BreakInheritanceList: BeforeColon

I prefer
AlignConsecutiveAssignments: true instead of false
ContinuationIndentWidth: 4 instead of 2


PointerAlignment: Left is fine if we do not allow declaration of multiple variables on one line.
AlwaysBreakTemplateDeclarations: Yes, should be true?

80 columns is something I push in my new developments so that we can have almost 2 files edited on common displays and almost 3 in wide screens. So convenient for developers and for Xdiffs, to achieve this 2 spaces indentation is the right number.

ssanadhya commented:
@lionelgo @rdefosse thanks for the comments!
I second Lionel on the 80 character limit as it makes it easy to work with multiple tabs on wide screens. I have made the following edits in the new version:

AlignConsecutiveAssignments: true
SpacesBeforeTrailingComments: 2 
ContinuationIndentWidth: 4

I have left the following unchanged
SpacesInContainerLiterals: false

Let me know what you think.

ssanadhya committed:
Fix clang-format file to position braces and parameters correctly (#1341)

Summary:
Pull Request resolved: https://github.com/facebookincubator/magma/pull/1341

We would like to follow the prevalent C++ style guidelines in our C/C++ code
formatting. This change fixes the clang-format style file as follows:
- In class/funtion definitions, put opening brace on the same line as the
  function/class name
- In function declarations/definitions/calls pack the parameter/arguments in as
  few lines as possible
- In class definitions, break line before the colon if the inheritance list
  doesn't fit in one line
- In constructor initialization, break line before the colon if the initializers
  don't fit in one line
- Horizontal whitespace fixes, such as space before trailing comments and using
  4 spaces for continuation lines
This is a preview diff to highlight the changes. Future diffs will apply the
format to entire oai/ directory.

Reviewed By: themarwhal, pshelar, koolzz

Differential Revision: D20499279

fbshipit-source-id: 1491bbd23dc4012e853ad08a311db1697c4af6b4

rdefosse commented:
All,
alone on the "80-char" no-limit... So I will differ to the majority 😞 
The rest is OK for me.

The last point I want to stress is when to apply to the whole repository.
In our experience, it might create huge merge conflicts on "big" pull requests (such as the one that Mohit will be doing soon).
I would like to double-check w/ Mohit if he can apply in this branch and see if merge conflicts are manageable,

But let discuss that later.

ssanadhya commented:
Good point @rdefosse, I will check with Mohit on how to avoid merge conflicts.

lionelgo commented:
This is fine for me, we could take also this for many of our CN repos.
